jwst/flatfield/flat_field.py

```diff
@@ -205,23 +206,28 @@ def apply_flat_field(science, flat, inverse=False):
 
     # Now let's apply the correction to science data and error arrays.  Rely
     # on array broadcasting to handle the cubes
+    flat_data_squared = flat_data * flat_data
     if not inverse:
         science.data /= flat_data
+        # NOTE: Re-arranging var_flat math will cause some NIRISS SOSS
+        # regression tests to fail with floating point differences.
+        science.var_flat = science.data ** 2 / flat_data_squared * flat_err ** 2
```

Comment:
I think I found the culprit for NIRISS failures. 😭 

RT running at https://github.com/spacetelescope/RegressionTests/actions/runs/14477166563